src/binary-encoding.adoc

```diff
@@ -0,0 +1,1600 @@
+= Binary Encoding
+:toc:
+
+[[encoding_primitives]]
+== Encoding primitives
+
+[[varuint]]
+=== `VarUInt`
```

Comment:
I'm open to renaming them. What would you suggest?

Reply:
What differentiates this from the 1.0 `Var(U)Int` is that this may be called "definite" (you know the byte count in the beginning) while 1.0 could be called "indefinite" (you only know you're done when you're done).

Reply:
Those proposals all emphasize the variable-width aspect, which is fine in this context (and I kinda like "flex").  However that aspect is not distinguishing WRT fully fledged integers (or symbols), so the terms don't work well as type names in the macro-signature DSL.   There, I'm currently considering `plain_int` _etc._

Reply:
> What differentiates this from the 1.0 Var(U)Int is that this may be called "definite" (you know the byte count in the beginning) while 1.0 could be called "indefinite" (you only know you're done when you're done).

My issue with this, is that all of the 1.1 encodings are definite.  You know the byte count for `Fixed...` integers as well.  Definite conveys something very specific about a difference of 1.0/1.1 variable length integers that I don't think stands on its own.